src/views/esp/view_esp.cpp

```diff
@@ -85,6 +85,85 @@ namespace big
 			if (distance < g.esp.box_render_distance[1] && distance > g.esp.box_render_distance[0] && g.esp.box)
 				draw_list->AddRect({esp_x - (62.5f * multplr), esp_y - (175.f * multplr)}, {esp_x - (62.5f * multplr) + (125.f * multplr), esp_y - (175.f * multplr) + (350.f * multplr)}, esp_color);
 
+			if (distance < g.esp.skeleton_render_distance[1] && distance > g.esp.skeleton_render_distance[0] && g.esp.skeleton)
+			{
+				std::vector<PedBones> bone_ids = // size: 18
+				{
+					PedBones::SKEL_Head, 
+					PedBones::SKEL_Neck_1,
+					PedBones::SKEL_Spine_Root, 
+					PedBones::SKEL_Pelvis, 
+					PedBones::SKEL_L_Thigh, 
+					PedBones::SKEL_R_Thigh, 
+					PedBones::SKEL_L_Calf, 
+					PedBones::SKEL_R_Calf, 
+					PedBones::SKEL_L_Foot, 
+					PedBones::SKEL_R_Foot, 
+					PedBones::SKEL_L_Clavicle, 
+					PedBones::SKEL_R_Clavicle, 
+					PedBones::SKEL_L_UpperArm, 
+					PedBones::SKEL_R_UpperArm, 
+					PedBones::SKEL_L_Forearm, 
+					PedBones::SKEL_R_Forearm, 
+					PedBones::SKEL_L_Hand, 
+					PedBones::SKEL_R_Hand
+				};
+
+				std::vector<rage::fvector4> bones = {};
+				const size_t skel_size            = 18;
+
+				for (int i = 0; i < skel_size; i++)
```

Comment:
use for (auto bone : bone_ids) and pass bone to get_ped_bone. You do not need or care about the index.

Reply:
Uhm no, I need to have it as a "for (int)" to get the position offset for "bones_screen[i]" as you can see in the next for-loop. Arguably, this one you highlighted could be switched, although i like to keep everything looking the same. So if Yimura wants it changed I will change it.

Reply:
Nothing you said made sense. The index has nothing to do with the size, as that comes from the bones vector. You can simply refactor the code like this and it will work the same way. 

```
for (auto bone : bone_ids)
{
	rage::fvector4 out = {};
	if (g_pointers->m_gta.m_get_ped_bone(plyr->get_ped(), out, bone))
	{
		bones.push_back(out);
	}
}
```

Reply:
Read what I said again